File: app/src/main/java/io/trewartha/positional/MainFragmentPagerAdapter.java
package io.trewartha.positional;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import io.trewartha.positional.compass.CompassFragment;
import io.trewartha.positional.position.PositionFragment;

public class MainFragmentPagerAdapter extends FragmentPagerAdapter {

    public MainFragmentPagerAdapter(FragmentManager fragmentManager) {
        super(fragmentManager);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new PositionFragment();
            case 1:
                return new CompassFragment();
            default:
                return new PositionFragment();
        }
    }

    @Override
    public int getCount() {
        return 2;
    }
}

File: app/src/main/java/io/trewartha/positional/MainFragmentPagerAdapter.kt
package io.trewartha.positional

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import io.trewartha.positional.compass.CompassFragment
import io.trewartha.positional.map.MapFragment
import io.trewartha.positional.position.PositionFragment

class MainFragmentPagerAdapter(fragmentManager: FragmentManager) : FragmentPagerAdapter(fragmentManager) {

    companion object {
        private val FRAGMENTS = arrayOf(
                MapFragment(),
                PositionFragment(),
                CompassFragment()
        )
    }

    override fun getItem(position: Int): Fragment {
        return FRAGMENTS[position]
    }

    override fun getCount(): Int {
        return FRAGMENTS.size
    }
}
